### ISSUE TITLE
Support multiple X and Y with keras graph api

### ISSUE THREAD

**Issue by bhack** [maxpumperla/elephas#16]
Actually Elephas doesn't account for multiple inputs/outputs.

**Comment by bhack** [comment]
Keras 1.0 is released. There is also a new [functional API for multi I/O](http://keras.io/getting-started/functional-api-guide/)

**Comment by NanaAkwasiAbayieBoateng** [comment]
I am having difficulty implementing the model below in elephas. From the Elephas page most of the examples combine all features into a single array and build the model.What if you want to enter multiple inputs into the  keras model such as below:


movie_embedding_size = user_embedding_size = 8

# Each instance consists of two inputs: a single user id, and a single movie id
user_id_input = keras.Input(shape=(1,), name='user_id')
movie_id_input = keras.Input(shape=(1,), name='movie_id')
user_embedded = keras.layers.Embedding(df.userId.max()+1, user_embedding_size, 
                                       input_length=1, name='user_embedding')(user_id_input)
movie_embedded = keras.layers.Embedding(df.movieId.max()+1, movie_embedding_size, 
                                        input_length=1, name='movie_embedding')(movie_id_input)

#from keras.layers import multiply
#output = multiply([user_embedded, movie_embedded])

#tf.keras.layers.multiplyx([user_embedded, movie_embedded])
#elementwise product
#keras.layers.Multiply
#dot product
dotted = keras.layers.Dot(2)([user_embedded, movie_embedded])
out = keras.layers.Flatten()(dotted)

model = keras.Model(
    inputs = [user_id_input, movie_id_input],
    outputs = out,
)
model.compile(
    tf.train.AdamOptimizer(0.001),
    
    loss='MSE',
    metrics=['MAE','MAPE'],
)

model.summary(line_length=88)




I would appreciate any help on converting this to distributed machine learning.Thanks

**Comment by maxpumperla** [comment]
@NanaAkwasiAbayieBoateng it's currently not possible, but I'll add this for the next release (maybe next month)

**Comment by NanaAkwasiAbayieBoateng** [comment]
Thank you very much. Looking forward to it.

**Comment by GeoffDuniam** [comment]
Hi @maxpumperla 

Any progress on this? We're interested on this functionality as well for classification of astronomical objects where we may have multiple feature arrays.

Cheers

Geoff
